components/movementsensor/gpsrtkserial/gpsrtkserial.go

```diff
@@ -662,3 +752,164 @@ func (g *rtkSerial) Close(ctx context.Context) error {
 	}
 	return nil
 }
+
+// sendGGAMessage sends GGA messages to the NTRIP Caster over a TCP connection
+// to get the NTRIP steam when the mount point is a Virtual Reference Station.
+func (g *rtkSerial) sendGGAMessage() error {
+	g.ntripMu.Lock()
+	defer g.ntripMu.Unlock()
+
+	if !g.isConnected {
+		g.readerWriter = g.connectToVirtualBase()
+	}
+
+	// read from the socket until we know if a successful connection has been
+	// established.
+	for {
+		if g.readerWriter.Reader == nil || g.readerWriter.Writer == nil {
```

Comment:
When would this happen? I don't see an obvious place where we'd ever mutate any fields within `g.readerWriter`, but maybe I missed it.